src/routes.rs

```diff
@@ -69,7 +69,90 @@ impl DatabaseAppState {
         terms
     }
 
-    fn courses(&self, term: Term) -> Result<Vec<ThinCourse>> {
+    fn courses(&self, term: Term, keys: &[&ThinCourse]) -> Result<Vec<Course>> {
```

Comment:
i'll leave @malcolmseyd to comment on the business logic here, i'm not really familiar with what this ideally would look like

Reply:
I wrote this so LGTM :laughing: :+1: